File: src/main/java/com/peasenet/mixins/MixinLivingEntity.java
package com.peasenet.mixins;

import com.peasenet.main.GavinsMod;
import net.minecraft.entity.LivingEntity;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;

@Mixin(LivingEntity.class)
public class MixinLivingEntity {
    @Inject(at = @At("RETURN"), method = "isClimbing", cancellable = true)
    public void onLadder(CallbackInfoReturnable<Boolean> cir) {
        if(GavinsMod.ClimbEnabled()) {
            boolean enabled =  ((LivingEntity) (Object) this).horizontalCollision;
            cir.setReturnValue(enabled);
        }
    }
}

File: src/main/java/com/peasenet/mixins/MixinLivingEntity.kt
/*
 * Copyright (c) 2022-2023. Gavin Pease and contributors.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
 * associated documentation files (the "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
 * of the Software, and to permit persons to whom the Software is furnished to do so, subject to the
 * following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
 * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
 * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT
 * OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */
package com.peasenet.mixins

import com.peasenet.main.GavinsMod.Companion.isEnabled
import com.peasenet.mods.Type
import net.minecraft.entity.LivingEntity
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable

/**
 * @author gt3ch1
 * @version 03-02-2023
 */
@Mixin(LivingEntity::class)
class MixinLivingEntity {
    @Inject(at = [At("RETURN")], method = ["isClimbing"], cancellable = true)
    fun onLadder(cir: CallbackInfoReturnable<Boolean?>) {
        if (isEnabled(Type.CLIMB)) {
            val enabled = (this as LivingEntity).horizontalCollision
            cir.returnValue = enabled
        }
    }
}